openssh-ccache-check/openssh-ccache-check.py

```diff
@@ -0,0 +1,114 @@
+        return pocket_version
+
+    def get_version(self, pocket):
+        try:
+            version = self.rmadison[pocket]
+        except KeyError:
+            version = None
+        return version
+
+    def get_latest_version_in_series(self, series):
+        latest_version = (None,None)
+        for pocket in [
+            series, f"{series}-updates", f"{series}-security",
+            f"{series}-proposed"]:
+            if pkg_version_greater_than(self.get_version(pocket), latest_version[0]):
+                latest_version = (self.get_version(pocket), pocket)
+        return latest_version
+
+        
+class CcachePPA(object):
+
+    def __init__(self, lp, ppa_owner, ppa_name, series="jammy", package="openssh"):
+        self.lp = lp
+        self.package = package
+        self.distro_series = self.lp.distributions["ubuntu"].getSeries(
+            name_or_version=series)
+        self.archive = self.lp.people[ppa_owner].getPPAByName(name=ppa_name)
+
+    def get_latest_version(self):
+        sources = self.archive.getPublishedSources(
+            source_name=self.package, distro_series=self.distro_series,
+            status="Published")
+        assert len(sources) == 1
+        return sources[0].source_package_version.strip()
+
+
+# avoiding a dependency on python3-apt
+def pkg_version_greater_than(v1, v2):
+    if not v1:
+        return False
+    if not v2:
+        return True
+    cmd = ["dpkg", "--compare-versions", v1, "gt", v2]
+    ret = call(cmd)
+    return ret == 0
+
+
+def main():
+    warn = False
+    lp = Launchpad.login_anonymously("ccache-check", "production")
+    openssh_ppa_proposed = CcachePPA(
+        lp, "canonical-server", "openssh-server-default-ccache-proposed")
+    openssh_ppa_release = CcachePPA(
+        lp, "canonical-server", "openssh-server-default-ccache")
+    openssh_ppa_testing = CcachePPA(lp, "ahasenack", "openssh-ccache-testing")
+    ppa_version = {}
+    ppa_version["testing"] = openssh_ppa_testing.get_latest_version()
+    ppa_version["proposed"] = openssh_ppa_proposed.get_latest_version()
+    ppa_version["release"] = openssh_ppa_release.get_latest_version()
+
+    openssh_archive = RmadisonPackage("openssh")
+    latest_version_in_archive = (
+        openssh_archive.get_latest_version_in_series("jammy"))
+
+    print(f"Latest version in jammy is {latest_version_in_archive}")
+    print()
+    for pocket,version in ppa_version.items():
+        print(f"Latest version in {pocket} ccache ppa is {version}")
+        if pkg_version_greater_than(latest_version_in_archive[0], version):
+            print(
+                f"WARNING: Latest version in archive "
+                f"({latest_version_in_archive[0]}) is higher than version "
+                f"{version} from ppa {pocket}")
+            warn = True
+        print()
+    if warn:
+        print("ACTION NEEDED")
+    else:
+        print("ALL GOOD")
+    return warn
+
+if __name__ == "__main__":
+    sys.exit(main())
```

Comment:
I found it slightly counterintuitive to rely on `bool(False) == 0` to exit with RC=0, but I see no actual issues with that.

Reply:
Yeah, that makes sense. Changing.